tests/handlers/test_user_directory.py

```diff
@@ -119,19 +124,22 @@ def test_handle_user_deactivated_support_user(self):
                 user_id=s_user_id, password_hash=None, user_type=UserTypes.SUPPORT
             )
         )
-
-        self.store.remove_from_user_dir = Mock(return_value=defer.succeed(None))
-        self.get_success(self.handler.handle_local_user_deactivated(s_user_id))
-        self.store.remove_from_user_dir.not_called()
+        with patch.object(
+            self.store, "remove_from_user_dir", return_value=defer.succeed(None)
+        ) as mock:
+            self.get_success(self.handler.handle_local_user_deactivated(s_user_id))
+        self.get_success(mock.not_called())
```

Comment:
Does this do what it's supposed to? [`Mock`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock) doesn't seem to have a `not_called` method?

Reply:
I don't think so. Has this been broken for a while then!?

Reply:
changing to `assert_not_called` makes the test fail! 

My reading: the test is trying to ensure that `handle_local_user_deactivated` doesn't try to delete a user from the db who would never have been in the directory. But I think `handle_local_user_deactivated` has never had any logic in it, so I think this test has been incorrectly passing forever.

We could make `handle_local_user_deactivated` call `is_excluded_from_user_directory`, but that involves a SELECT from the db anyway to see if they're deactivated or a support user. I'm guessing that's the same order of magnitude of delay as doing a no-op deletion? And so I think the current behaviour is fine.

I guess it's worth having this kind of test here though, to make sure that synapse doesn't fall over when we deactivate a support user?

Reply:
> I guess it's worth having this kind of test here though, to make sure that synapse doesn't fall over when we deactivate a support user?

Additional: maybe test_user_directory isn't the right place for such a test.